lib/spack/spack/audit.py

```diff
@@ -417,8 +442,6 @@ def _unknown_variants_in_dependencies(pkgs, error_cls):
 @package_directives
 def _version_constraints_are_satisfiable_by_some_version_in_repo(pkgs, error_cls):
     """Report if version constraints used in directives are not satisfiable"""
-    import spack.repo
-
```

Comment:
For what is worth, all these imports were local on purpose, since they insist on different parts of Spack and they document which imports are needed for the test being run.

Reply:
I guess I’m not seeing the benefit of this — they made the code hard to read for me and some of them (llnl.util Ines)  aren’t really documenting what part of spack is used — just what’s needed for the code in the test.